control-plane/api-gateway/binding/validation.go

```diff
@@ -431,6 +431,53 @@ func routeAllowedForListenerHostname(hostname *gwv1beta1.Hostname, hostnames []g
 	return false
 }
 
+func keyFromLocalObjectReference(reference *gwv1beta1.LocalObjectReference) string {
+	return string(reference.Kind) + string(reference.Name)
+}
+
+// externalRefsOnRouteAllExist
+func externalRefsOnRouteAllExist(route *gwv1beta1.HTTPRoute, resources *common.ResourceMap) bool {
+	extensionRefs := resources.GetHTTPRouteExtensionRefs(route)
+	if len(extensionRefs) == 0 {
+		return true
+	}
+
+	for _, rule := range route.Spec.Rules {
+		for _, filter := range rule.Filters {
+			if filter.ExtensionRef == nil {
+				continue
+			}
+			filterKey := keyFromLocalObjectReference(filter.ExtensionRef)
+
+			_, ok := extensionRefs[filterKey]
+			if !ok {
+				//filter doesn't exist
+				return false
+			}
+
+		}
+	}
+
+	return true
+}
+
+// externalRefsKindAllowedOnRoute
+func externalRefsKindAllowedOnRoute(route *gwv1beta1.HTTPRoute) bool {
+	for _, rule := range route.Spec.Rules {
+		for _, filter := range rule.Filters {
+			if filter.ExtensionRef == nil {
+				continue
+			}
+
+			if filter.ExtensionRef.Kind != "HTTPRouteRetryFilter" {
```

Comment:
Can we use a constant here? Also wondering if we can collapse this with the above method into something like a `validateFilterExtensions` call? That way we just iterate over the rules/filters once and then return an error value corresponding to either an invalid filter type of the filter not existing?